google/cloud/pubsub/internal/tracing_pull_ack_handler.cc

```diff
@@ -38,36 +38,39 @@ using Attributes =
 
 namespace {
 
-auto CreateLinks() {
+/// Create a list of links with @p span context if compiled with open telemetery
+/// ABI 2.0.
+auto CreateLinks(
+    std::shared_ptr<opentelemetry::trace::SpanContext> const& span_context) {
```

Comment:
passing a `shared_ptr<>` as a `const&` is a code smell.  You probably want:

```suggestion
auto CreateLinks(opentelemetry::trace::SpanContext const& span_context) {
```

Reply:
Done